File: discovery-service/src/main/java/com/sudo248/discoveryservice/controller/dto/SupplierProductDto.java
package com.sudo248.discoveryservice.controller.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SupplierProductDto {
    private String supplierId;
    private String productId;
    private RouteDto route;
    private int amountLeft = 0;
    private double price = 0.0;
    private int soldAmount = 0;
    private double rate = 0.0;
}

File: client/SocStaff/app/src/main/java/com/sudo248/soc_staff/data/dto/discovery/SupplierProductDto.kt
package com.sudo248.soc_staff.data.dto.discovery

import com.sudo248.soc_staff.domain.entity.discovery.Route

data class SupplierProductDto(
    val supplierId: String,
    val productId: String,
    val route: Route,
    val amountLeft: Int,
    val price: Double,
    val soldAmount: Int,
    val rate: Double,
)
